test/unit/config/test__base.py

```diff
@@ -275,45 +275,60 @@ def test_container_volume_mounting_with_Z(mock_isdir, mock_exists, mock_makedirs
 
 
 @pytest.mark.parametrize('container_runtime', ['docker', 'podman'])
-def test_containerization_settings(tmpdir, container_runtime):
-    with patch('ansible_runner.config._base.BaseConfig.containerized', new_callable=PropertyMock) as mock_containerized:
-        rc = BaseConfig(private_data_dir=tmpdir)
-        rc.ident = 'foo'
-        rc.cmdline_args = ['main.yaml', '-i', '/tmp/inventory']
-        rc.command = ['ansible-playbook'] + rc.cmdline_args
-        rc.process_isolation = True
-        rc.runner_mode = 'pexpect'
-        rc.process_isolation_executable = container_runtime
-        rc.container_image = 'my_container'
-        rc.container_volume_mounts = ['/host1:/container1', 'host2:/container2']
-        mock_containerized.return_value = True
-        rc.execution_mode = BaseExecutionMode.ANSIBLE_COMMANDS
-        rc._prepare_env()
-        rc._handle_command_wrap(rc.execution_mode, rc.cmdline_args)
+def test_containerization_settings(tmp_path, container_runtime, mocker):
+    mocker.patch.dict('os.environ', {'HOME': str(tmp_path)}, clear=True)
```

Comment:
Have you considered using the built-in `monkeypatch.setenv()`?

https://docs.pytest.org/en/latest/how-to/monkeypatch.html

Reply:
I haven't yet figured out when to use `monkeypath` vs `mocker.path`. I still have things to learn. :)

Reply:
out of the 3 options outlined here:

https://github.com/pytest-dev/pytest/issues/4576#issuecomment-449864333

I have seen a lot converging on option (2), because python2 tends to be in the rear view mirror by now. The pattern `from unittest import mock` already exists in ansible-runner unit tests. However, the pytest-mock you're adding here is more naturally aware of the pytest test scope, which means you don't have to indent everything in context managers. Thus, I raise no objections to it.

Reply:
> The pattern from unittest import mock already exists in ansible-runner unit tests.

I have a stashed commit locally that removes all of that in favor of the `pytest-mock` plugin. It ended up being pretty huge and took me several hours to get working which is why I didn't include those changes in this PR. I'll do that in (yet another) PR.

Reply:
I had a good talk with @webknjaz about `monkeypach` vs `mocker.patch` and I have a better idea of when to use each now. `monkeypatch` doesn't have the ability to clear the environment, but I'm not sure that is strictly necessary. I will try it out see if it works.

Reply:
_Technically_, `monkeypatch.setattr(os, 'environ', {'HOME': str(tmp_path)})` would probably emulate this. But I agree, it's usually unnecessary to clear all the other unnecessary env vars. If you expect certain env vars not to be set, use `monkeypatch.delenv()` for that.

Reply:
> out of the 3 options outlined here:
> 
> [pytest-dev/pytest#4576 (comment)](https://github.com/pytest-dev/pytest/issues/4576#issuecomment-449864333)
> 

I think that what Ronny points out in the very next comment is even more important to take into account: https://github.com/ansible/ansible-runner/pull/815#discussion_r707453680.

TL;DR If you end up using any sort of mocking technique, you probably have an architectural problem. If you were to just use dependency injection, there would be exactly zero need for this.

Reply:
> If you end up using any sort of mocking technique, you probably have an architectural problem. If you were to just use dependency injection, there would be exactly zero need for this.

I absolutely agree with this. I am huge fan of test driven development for this reason, among others. Currently, I'm just trying to improve the state of the tests. Eventually I hope to improve the code itself.